.dockerignore

```diff
@@ -0,0 +1,5 @@
+**/.git
+**/.gitmodules
+**/.travis.yml
+**/Dockerfile
+**/*.md
```

Comment:
Yeah, there's probably a lot more stuff that could be excluded. Next month's release of Docker includes a feature called "multi-stage builds" (check it out if you haven't; it's cool!) which will let us perform a build and then throw away the entire source tree, leaving only the built artifacts. Since that's coming so soon and it'll totally solve this problem, I figured it's not worth the effort of optimizing this too much now.

On the other there, now that I think about it, even with multi-stage builds there *is* some value, since reducing the size of the context reduces the amount of data that Docker needs to copy into the image and could eliminate some spurious rebuilds. So perhaps a followup PR that added a more thorough .dockerignore would be worthwhile. Better to land this first, though, since any changes would require retesting of the downstream Docker images.